### PR TITLE
Responsive Curve2

### LINKED ISSUE #272: Responsive theme
Hey everyone, this is my first post on github :)

Anyway, I have a suggestion  -  SMFs default theme to be responsive.

We all know that Responsive web design is the best way to handle support for various screen sizes.

By the way, I could help with front-end development. 
Can someone tell me - is there a specified design for SMF 2.1? Can I make a suggestion?

### LINKED ISSUE #347: Mobile/Tablet Friendly UI
I like to do some of the administration on my Android tablet (I can hover, with my S Pen), but it won't go to the sub-pages. So I have to enable the side-bar. Could you implement something so it's possible to use touch devices since they are becoming so very common?

### PR DESCRIPTION
**Pull request by XinYenFon**
This PR gives SMF 2.1 to handle some resolutions to display properly (aka Responsive). Admin side currently untouched and probably stay that way for now.
##### Tested with:
- Nokia Lumia 520
- Samsung Galaxy S3
- Samsung Galaxy Note2
- Samsung Galaxy S4
- Nexus7

Fixes #272
Fixes #347
Signed-off-by: Antes antes@simplemachines.org

### CODE REVIEW

**Comment by MissAllSunday** [comment]
@Antes Do you wish to merge this now or do you want to add something else before merging it?

**Comment by XinYenFon** [comment]
Actually i opened this PR for tracking and live testing with people. It currently has dependencies;

https://github.com/SimpleMachines/SMF2.1/pull/1551
https://github.com/SimpleMachines/SMF2.1/pull/1477
https://github.com/SimpleMachines/SMF2.1/pull/1471

**Comment by MissAllSunday** [comment]
I'm sorry, don't fully understand what you mean. I thought this PR had the bulk of the responsive theme and will need to be merged first before merging any other PRs.

**Comment by XinYenFon** [comment]
Yes and no, its full work for responsive theme. No, once we change the parts of the theme, we need to update the CSS file as well. We are doing the responsive side with little less template addition and leaving all work to index.responsive.css.

For example; one of the PR changes table based styling to div based for board-index so we need to change CSS file to newly introduced systematic/classes to keep it working.

**Comment by MissAllSunday** [comment]
I'm sorry, it is still unclear for me if this branch needs to be merged or not or if the other ones needs to be merged and if so in what order to a void unnecessary merge conflicts.

**Comment by XinYenFon** [comment]
#1471 & #1551 already merged in, if it has no merge conflict first #1477 needs to be pull in then I'll change some stuff in this PR and it'll be ready for merge.

File: Themes/default/index.template.php [hunk 1]
```diff
@@ -99,7 +99,8 @@ function template_html_above()
 	<meta http-equiv="Content-Type" content="text/html; charset=', $context['character_set'], '">
 	<meta name="description" content="', !empty($context['meta_description']) ? $context['meta_description'] : $context['page_title_html_safe'], '">', !empty($context['meta_keywords']) ? '
 	<meta name="keywords" content="' . $context['meta_keywords'] . '">' : '', '
-	<title>', $context['page_title_html_safe'], '</title>';
+	<title>', $context['page_title_html_safe'], '</title>
+	<meta name="viewport" content="width=device-width, initial-scale=1" />';
```
Comment (tekkla) [inline Themes/default/index.template.php]: Only ">"

Comment (XinYenFon) [inline Themes/default/index.template.php]: Yea html5 I'll fix it when I had time thanks rookie mistake :P

-----Original Message-----

File: Themes/default/index.template.php [hunk 2]
```diff
@@ -407,6 +408,11 @@ function template_menu()
 
 	echo '
 				<div id="main_menu">
+					<a class="button mobile">Main Menu</a>
+					<script>
+						$( "a.button.mobile" ).click(function() {
+							$( "#menu_nav" ).toggle();
+						});</script>
```
Comment (tekkla) [inline Themes/default/index.template.php]: Maybe you want to shift inline scripts to a js file. It is quite easier to maintain.

File: Themes/default/index.template.php (hunk 1 above)
Comment (tekkla) [inline Themes/default/index.template.php]: You gave me the link ;)

File: Themes/default/index.template.php (hunk 2 above)
Comment (XinYenFon) [inline Themes/default/index.template.php]: jQ toggle only for some tests I don't see we keep it for final. We have some different ideas but maybe too soon to talk about it :)

**Pull request merged**